src/edu/cornell/mannlib/vitro/webapp/edit/n3editing/configuration/generators/DatasetLandingPageGenerator.java

```diff
@@ -0,0 +1,205 @@
+        // we do not allow editing.
+        setSparqlQueries(editConfiguration, vreq);
+
+        // set the field names that are used in the form. This tells VIVO the form names, whether they are required or not, and if a Literal you can
+        // tell it what type of literal
+        setFields(editConfiguration, vreq);
+
+        // This tells the configuration which freemarker template file to use to generate the form. In our form we have only one text box for the landingPage.
+        editConfiguration.setTemplate("addLandingPage.ftl");
+
+
+        // This does basic validation on all of the fields, if they are required. This is a default validator provided by VIVO
+        editConfiguration.addValidator(new AntiXssValidation());
+
+        // Adding additional data, specifically edit mode, to be used by the freemarker template (ftl)
+        addFormSpecificData(editConfiguration, vreq);
+
+        // if we had any pre-processing to do on any of the fields in the form we would use the AddLandingPagePreprocessor object. Right now it does nothing.
+        /*
+        editConfiguration.addEditSubmissionPreprocessor(
+                new AddLandingPagePreprocessor(editConfiguration));
+        */
+
+        prepare(vreq, editConfiguration);
+        return editConfiguration;
+    }
+
+    // sets the variable names for the subject, predicate and object use in the generation of the N3
+    private void setVarNames(EditConfigurationVTwo editConfiguration) {
+        editConfiguration.setVarNameForSubject("dataset");
+        editConfiguration.setVarNameForPredicate("landingPage");
+        editConfiguration.setVarNameForObject("uri");
+    }
+
+    // Generate the N3 that will be used to create the new triples. ?uri is replaced with <uri> from the form as it is a uri. We make that entity a
+    // foaf:Document so that it appears properly in the dataset display
+    private List<String> generateN3Required() {
+        return list(getLandingPageN3());
+    }
+
+    // what we do is we have the user specify a URL that is used to access datasets, not a download link). We then use that URL as the URI of a
+    // foaf:Document. In the custom display we use the URI of the foaf:Document in the display that links to that URI. We do not set any
+    // properties for the foaf:Document itself.
+    private String getLandingPageN3() {
+        String ret = "@prefix foaf: <" + foaf + "> . " +
+                "@prefix dcat: <" + dcat + "> ." +
+                "?dataset dcat:landingPage ?uri ." +
+                "?uri a foaf:Document ." ;
+        return ret ;
+    }
+
+    // if we had any new resources, like creating a new object or a new sub object, then we would do that here
+    private Map<String, String> generateNewResources(VitroRequest vreq) {
+        String DEFAULT_NS_TOKEN=null; //null forces the default NS
```

Comment:
DEFULAT_NS_TOKEN is not used in this case. Can be removed.

The only reason I kept the methods there is because we'll need them in other custom forms. Though we could just look at other code for that, like our project update code. In fact, we can just not call that method in getEditConfiguration. I'll take a look.